akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpMessage.scala

```diff
@@ -284,6 +304,10 @@ final class HttpRequest(
   override def isRequest = true
   override def isResponse = false
 
+  @deprecated("use the constructor that includes an attributes parameter instead", "10.2.0")
```

Comment:
I guess, we should propose to start with a simple request/response and use the `withX` more. That will also be the more compatible way going forward.